crates/cow-amm/src/maintainers.rs

```diff
@@ -0,0 +1,80 @@
+use {
+    crate::{cache::Storage, Amm},
+    contracts::ERC20,
+    ethcontract::futures::future::{join_all, select_ok},
+    ethrpc::Web3,
+    shared::maintenance::Maintaining,
+    std::sync::Arc,
+    tokio::sync::RwLock,
+};
+
+pub struct EmptyPoolRemoval {
+    storage: Arc<RwLock<Vec<Storage>>>,
+    web3: Web3,
+}
+
+impl EmptyPoolRemoval {
+    pub fn new(storage: Arc<RwLock<Vec<Storage>>>, web3: Web3) -> Self {
+        Self { storage, web3 }
+    }
+
+    /// Checks if the given AMM has a zero token balance.
+    async fn check_single(&self, amm: Arc<Amm>) -> bool {
```

Comment:
nanonit: more expressive method name

```suggestion
    async fn has_zero_balance(&self, amm: Arc<Amm>) -> bool {
```